target/linux/lantiq/dts/EASY220.dtsi

```diff
@@ -0,0 +1,295 @@
+	};
+
+	gphy-xrx200 {
+		compatible = "lantiq,phy-xrx200";
+		firmware = "lantiq/vr9_phy22f_a2x.bin";
+		phys = [ 00 01 ];
+	};
+
+//	gpio-keys-polled {
+//		compatible = "gpio-keys-polled";
+//		#address-cells = <1>;
+//		#size-cells = <0>;
+//		poll-interval = <100>;
+/*		reset {
+			label = "reset";
+			gpios = <&gpio 7 GPIO_ACTIVE_HIGH>;
+			linux,code = <0x198>;
+		};*/
+//	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		power_a: power_amber {
+			label = "power_amber";
+			gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+		};
+		power_g: power_green {
+			label = "power_green";
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		};
+		lan_amber {
+			label = "lan_amber";
+			gpios = <&gpio 33 GPIO_ACTIVE_LOW>;
+		};
+		lan_g: lan_green {
+			label = "lan_green";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+		dsl_amber {
+			label = "dsl_amber";
+			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+		};
+		dsl: dsl_green {
+			label = "dsl_green";
+			gpios = <&gpio 36 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&gpio {
+	pinctrl-names = "default";
+	pinctrl-0 = <&state_default>;
+
+	state_default: pinmux {
+		mdio {
+			lantiq,groups = "mdio";
+			lantiq,function = "mdio";
+		};
+		phy-rst {
+			lantiq,pins = "io42";
+			lantiq,pull = <0>;
+			lantiq,open-drain = <0>;
+			lantiq,output = <1>;
+		};
+		exin {
+			lantiq,groups = "exin3";
+			lantiq,function = "exin";
+		};
+		/*stp {
+			lantiq,groups = "stp";
+			lantiq,function = "stp";
+		};*/
+		/*nand {
+			lantiq,groups = "nand cle", "nand ale",
+					"nand rd", "nand rdy";
+			lantiq,function = "ebu";
+		};*/
+		/*pci {
+			lantiq,groups = "gnt1", "req1";
+			lantiq,function = "pci";
+		};*/
+		/*led {
+			lantiq,groups = "led";
+			lantiq,function = "led";
+		};*/
+		conf_in {
+			lantiq,pins = "io39"; /* exin3 */
+			lantiq,pull = <2>;
+		};
+		conf_out {
+			lantiq,pins =  // "io21", "io38", /* pcie */
+//					"io12", "io15", /* annexa, annexb */
+					"io0", "io1", "io8", "io11", "io33", "io36"; /* led io34, io35 */
+			lantiq,open-drain = <0>;
+			lantiq,pull = <0>;
+			lantiq,output = <1>;
+		};
+	};
+	pins_spi_default: pins_spi_default {
+		spi_in {
+			lantiq,groups = "spi_di";
+			lantiq,function = "spi";
+		};
+		spi_out {
+			lantiq,groups = "spi_do", "spi_clk", "spi_cs4";
+			lantiq,function = "spi";
+			lantiq,output = <1>;
+		};
+	};
+};
+
+&spi {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pins_spi_default>;
+
+	status = "ok";
+
+	m25p80@4 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "winbond,w25q64", "jedec,spi-nor";
+		reg = <4 0>;
+		linux,modalias = "m25p80", "w25q64";
+		linux,mtd-name = "ltq_sflash";
+		spi-max-frequency = <10000000>;
+		m25p,fast-read;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				reg = <0x0 0x20000>;
+				label = "uboot";
+				read-only;
+			};
+
+			partition@20000 {
+				reg = <0x20000 0x10000>;
+				label = "gphyfirmware";
+				read-only;
+			};
+
+			partition@30000 {
```

Comment:
please merge kernel, firmware and rootfs into a single big firmware partition. Remove the read-only property from that partition, otherwise neither the config can be saved nor an update performed.